functionary/ui/views/function.py

```diff
@@ -102,6 +104,18 @@ def execute(request: HttpRequest) -> HttpResponse:
                     code="invalid",
                 ),
             )
+        except MaxRetryError as err:
+            form.add_error(
+                None,
+                ValidationError(
+                    f"{err.reason}. Unable to submit files.", code="invalid"
+                ),
+            )
+        except S3Error as err:
+            form.add_error(
+                None,
+                ValidationError(f"{err.message}.", code="invalid"),
+            )
 
     args = {"form": form, "function": func}
     return render(request, "core/function_detail.html", args)
```

Comment:
We should update this to include a status code and then set the status code appropriately in the exception handling.  It's a little weird to run into the S3 related errors, for example, but have the HTTP response still be 200.  For ValidationError the response should be 400.  For the S3 errors I'd probably say 503.

Reply:
Fixed, but just for the Function detail view. I'll add the status codes to the other views that spawn Tasks as I add support for file parameters for those views